airflow/jobs/backfill_job.py

```diff
@@ -629,6 +629,7 @@ def _per_task_process(task, key, ti, session=None):  # pylint: disable=too-many-
             _dag_runs = ti_status.active_runs[:]
             for run in _dag_runs:
                 run.update_state(session=session)
+                session.merge(run)
```

Comment:
Is this change related to scheduler HA?

Reply:
It is, yes, but I can do this better

In [dagrun.py](https://github.com/apache/airflow/pull/10956/files#diff-32aa8dbb910719ef24a39cab5d0f2a97L376-L379) I removed the commit (on purpose) but also removed the merge by mistake:

```diff
-        # todo: determine we want to use with_for_update to make sure to lock the run
-        session.merge(self)
-        session.commit()
-
```

->


```diff
-        # todo: determine we want to use with_for_update to make sure to lock the run
        session.merge(self)
-        session.commit()
-
```

Reply:
Done now. all changes removed from this fie.